### PR TITLE
Design

### PR DESCRIPTION
**Pull request by danialm**

### CODE REVIEW

File: docs/reference/design/actions.md [hunk 1]
```diff
@@ -59,16 +59,15 @@ the same exact path for an action that needs to respond to all those verbs in th
...
 on the name of your enclosing endpoint definition class, in this case
 '/blogs' since our class is called `Blogs`. You can, however, override the prefix for a single route by prepending '//' to the path (like in the example above) if you don't want the resource-wide prefix to apply. Alternately, you can provide a special prefix of either `''` or `'//'` in the routing block to clear the prefix for any other paths given.
 
-*Note*: The above 'resetting' behavior of '//' applies *only* to any Resource-level route prefixes that may be defined. It will *not* override an API-wide `base_path` if one is defined (see [Global Api Info](global-api-information/)).
+*Note*: The above 'resetting' behavior of '//' applies *only* to any Resource-level route prefixes that may be defined. It will *not* override an API-wide `base_path` if one is defined (see [Global Api Info](api-definition#global-information)).
 
-You can inspect the complete Praxis routing table using `praxis routes` or `rake praxis:routes`:
+You can inspect the complete Praxis routing table using `bundle exec praxis routes` or `bundle exec rake praxis:routes`:
```
Comment (danialm) [inline docs/reference/design/actions.md:70]: I am not sure if want to include `bundle exec ` here, but it doe snot work w/o it for me

File: docs/reference/design/actions.md [hunk 2]
```diff
@@ -93,14 +92,14 @@ action :wildcards do
...
     attribute :splat, Attributor::Collection.of(String)
   end
 end
 ```
 
-Notice in the example above that if we use wilcard operators for our routes, we will also need to declare the
-`:splat` parameter in our action definition. This parameter will contain a collection of strings matching every wildcard in our route (and yes, you can have a route with multiple wildcards). If only one wildcard is used, `:splat` will still be an array, and will contain a single string element in it. See the [Mustermann site](https://github.com/rkh/mustermann) for more information about pattern types and other supported options.
+Notice in the example above that if we use wildcard operators for our routes, we will also need to declare the
+`:splat` parameter in our action definition. This parameter will contain a collection of strings matching every wildcard in our route (and yes, you can have a route with multiple wildcards). If only one wildcard is used, `:splat` will still be an array, and will contain a single string element in it. See the [Mustermann site](https://github.com/sinatra/mustermann) for more information about pattern types and other supported options.
```
Comment (danialm) [inline docs/reference/design/actions.md:102]: Looks like they have moved it 🤷‍♂️

File: docs/reference/design/actions.md [hunk 3]
```diff
@@ -180,10 +179,10 @@ It is common practice (especially in RESTful APIs) to be able to accept incoming
 
 In other words: Praxis will inject a `:reference` parameter to the payload, pointing to the defined default MediaType of the endpoint (refer to [MediaType](media-types) for more info). It is for this reason that the following `create` action payload definition is enough if the default MediaType of the corresponding `Post` resource has those same attribute names defined.
 
-```
+```ruby
 action :create do
   routing { post '' }
-  payload do
+  payload reference: MediaTypes::Post do
```
Comment (danialm) [inline docs/reference/design/actions.md:185]: I think we need this here, right?

File: docs/reference/design/endpoints.md [hunk 4]
```diff
@@ -101,16 +99,16 @@ class Blogs
 end
 ```
 
-A MediaType in Praxis is often more than just an Internet media-type string.
+A MediaType in Praxis is often more than just an internet media-type string.
```
Comment (danialm) [inline docs/reference/design/endpoints.md:102]: "Internet" is definitely correct, it just looks a tad out dated :)

File: docs/reference/design/endpoints.md [hunk 5]
```diff
@@ -238,8 +235,21 @@ Having the appropriate `canonical_path` allows you to both generate and parse hr
   * `EndpointDefinition.to_href(<named arguments hash>)` to generate an href for the resource.
   * `EndpointDefinition.parse_href(<href String>)` to get a type-coerced hash of the parameters for the canonical action from the given string.
 
-Given a controller (class or instance), you can use use those helpers by first calling its its `definition` method to retrieve the `EndpointDefinition` it implements, and then using either `to_href` or `parse_href` as described above.
+Given a controller (class or instance), you can use use those helpers by first calling its `definition` method to retrieve the `EndpointDefinition` it implements, and then using either `to_href` or `parse_href` as described above. For example:
 
+```ruby
```
Comment (danialm) [inline docs/reference/design/endpoints.md:240]: I added this example here to clarify how to use them. I am not sure if it's a proper example. But I think we need one

File: docs/reference/design/actions.md (hunk 1 above)
Comment (blanquer) [reply docs/reference/design/actions.md:70]: For the rake it is good to add...for the binary it shouldn't be needed...

File: docs/reference/design/actions.md (hunk 2 above)
Comment (blanquer) [reply docs/reference/design/actions.md:102]: 👍

File: docs/reference/design/actions.md (hunk 3 above)
Comment (blanquer) [reply docs/reference/design/actions.md:185]: I think the default mt is used as a reference if there is one...but I need to double check

Comment (blanquer) [reply docs/reference/design/actions.md]: I've confirmed that's the case. I.e., if there is not reference option in the payload, Praxis will automatically add the default mediatype as a reference (if there is a default mediatype defined). 

So, we should delete the reference in the docs you added.

File: docs/reference/design/endpoints.md (hunk 5 above)
Comment (blanquer) [reply docs/reference/design/endpoints.md:240]: 👍

**Pull request merged**